### PR TITLE
Add end-to-end test to snap

### LINKED ISSUE #3: Add tests to show users how to test snaps
Right now we don't have any tests in this repository. To help users to get an understanding of how to test snaps, we should add some basic tests, at least for the snap.

I'm not sure if tests for the frontend are necessary, but we could add some anyway, to make it easier for users.

### PR DESCRIPTION
**Pull request by Mrtenz**
Closes #3.

This adds end-to-end tests to the snap using `@metamask/snaps-jest`.

### CODE REVIEW

**Comment by Mrtenz** [comment]
@SocketSecurity ignore-all

File: packages/snap/src/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { installSnap } from '@metamask/snaps-jest';
+import { expect } from '@jest/globals';
+import { panel, text } from '@metamask/snaps-ui';
+
+describe('onRpcRequest', () => {
+  it('hello', async () => {
```
Comment (FrederikBolding) [inline packages/snap/src/index.test.ts:6]:
```suggestion
  it('supports the "hello" method', async () => {
```
or something similar?

File: yarn.lock [hunk 2]
```diff
@@ -116,6 +165,18 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@babel/generator@npm:^7.22.5, @babel/generator@npm:^7.7.2":
```
Comment (FrederikBolding) [inline yarn.lock:168]: Can we dedupe?

File: yarn.lock [hunk 3]
```diff
@@ -1514,146 +1818,1517 @@ __metadata:
...
+  resolution: "@chakra-ui/control-box@npm:2.0.13"
+  peerDependencies:
+    "@chakra-ui/system": ">=2.0.0"
+    react: ">=18"
+  checksum: e0cc5d4ca804751b507ec4acf9605c44df18522117bcd02684513689ad64bf8a44853153977c976aed386e210c6dc94c386296fbed7da4ee4430c63a293106dc
   languageName: node
   linkType: hard
 
-"@ethersproject/abstract-signer@npm:^5.7.0":
+"@chakra-ui/counter@npm:2.0.14":
```
Comment (FrederikBolding) [inline yarn.lock:1994]: It seems less than ideal that we have to pull in Chakra to do E2E 🤔

File: packages/snap/package.json [hunk 4]
```diff
@@ -23,21 +23,24 @@
...
   "devDependencies": {
+    "@jest/globals": "^29.5.0",
     "@lavamoat/allow-scripts": "^2.0.3",
     "@metamask/auto-changelog": "^2.6.0",
     "@metamask/eslint-config": "^10.0.0",
     "@metamask/eslint-config-jest": "^10.0.0",
     "@metamask/eslint-config-nodejs": "^10.0.0",
     "@metamask/eslint-config-typescript": "^10.0.0",
     "@metamask/snaps-cli": "^0.32.2",
+    "@metamask/snaps-jest": "^0.35.2-flask.1",
```
Comment (FrederikBolding) [inline packages/snap/package.json:43]: Should we be using the Flask versions for the CLI too? 🤔

File: .github/workflows/build-lint-test.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          key: snap-${{ runner.os }}-${{ github.sha }}
+      - run: yarn install --immutable
+      - name: Install Google Chrome
+        run: yarn install-chrome
+      - name: Run e2e tests
+        run: yarn workspace snap run test
+
+  all-jobs-pass:
+    name: All jobs pass
+    runs-on: ubuntu-20.04
```
Comment (FrederikBolding) [inline .github/workflows/build-lint-test.yml:65]:
```suggestion
    runs-on: ubuntu-latest
```

File: packages/snap/src/index.test.ts (hunk 1 above)
Comment (Mrtenz) [reply packages/snap/src/index.test.ts:6]: Oops, this was intended to be a `describe` block 😅

File: packages/snap/package.json (hunk 4 above)
Comment (Mrtenz) [reply packages/snap/package.json:43]: I don't think there's any significant difference between `0.32.2` and `0.35.2-flask.1`?

File: yarn.lock (hunk 3 above)
Comment (Mrtenz) [reply yarn.lock:1695]: I realised that too. Probably should move `snaps-simulator` dependencies to `devDependencies`?

Comment (Mrtenz) [reply yarn.lock:1695]: It's not an issue with the template snap at least.

File: packages/snap/package.json (hunk 4 above)
Comment (FrederikBolding) [reply packages/snap/package.json:43]: There are a few nice things, like an updated Babel config: https://github.com/MetaMask/snaps/blob/main/packages/snaps-cli/CHANGELOG.md

But its more of a process question, should the template use the latest Flask release of CLI or the latest non Flask 🤔

**Review by ziad-saab — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**